src/stream/src/executor/exchange/permit.rs

```diff
@@ -45,17 +50,48 @@ pub fn channel(
     let permits = Arc::new(Permits { records, barriers });
 
     let max_chunk_permits: usize = initial_permits - batched_permits;
+    let metric_memory_size =
+        MEMORY_SIZE.with_label_values(&[&fragment_ids.0.to_string(), &fragment_ids.1.to_string()]);
+    let metric_num_rows =
+        NUM_ROWS.with_label_values(&[&fragment_ids.0.to_string(), &fragment_ids.1.to_string()]);
 
     (
         Sender {
             tx,
             permits: permits.clone(),
             max_chunk_permits,
+            metric_memory_size: metric_memory_size.clone(),
+            metric_num_rows: metric_num_rows.clone(),
+        },
+        Receiver {
+            rx,
+            permits,
+            metric_memory_size,
+            metric_num_rows,
         },
-        Receiver { rx, permits },
     )
 }
 
+static MEMORY_SIZE: LazyLock<IntGaugeVec> = LazyLock::new(|| {
+    register_int_gauge_vec_with_registry!(
```

Comment:
Better to use guarded version.

Reply:
I double-checked the guarded version. The guard will remove the label on drop. But for this case, multiple metrics may share the same label `[up_fragment_id, down_fragment_id]`. Thus any one of them drops will result in the label to be cleared. Therefore, guarded metrics should not be used here.

Reply:
It sounds like we need to find out another approach to clean up the entry when either of the fragment is dropped.